apps/server/src/api-data/db/migration/db.migration.v3.ts

```diff
@@ -0,0 +1,410 @@
+    const oldSettings = structuredClone(jsonData.settings) as old_Settings;
+
+    const migrated: Settings = { ...oldSettings, version: '4.0.0' };
+    return migrated;
+  }
+}
+
+type old_ViewSettings = {
+  dangerColor: string;
+  normalColor: string;
+  overrideStyles: boolean;
+  warningColor: string;
+  freezeEnd: boolean;
+  endMessage: string;
+};
+
+/**
+ * migrates a view settings from v3 to v4.0.0
+ * - drop `freezeEnd`
+ * - drop `endMessage`
+ */
+export function migrateViewSettings(jsonData: object): ViewSettings | undefined {
+  if (is.objectWithKeys(jsonData, ['viewSettings']) && is.object(jsonData.viewSettings)) {
+    const { dangerColor, normalColor, overrideStyles, warningColor } = structuredClone(
+      jsonData.viewSettings,
+    ) as old_ViewSettings;
+    return { dangerColor, normalColor, overrideStyles, warningColor };
+  }
+}
+
+type old_URLPreset = {
+  enabled: boolean;
+  alias: string;
+  pathAndParams: string;
+}[];
+
+/**
+ * migrates a url presets from v3 to v4
+ * - pathAndParams split into a target and search
+ */
+export function migrateURLPresets(jsonData: object): URLPreset[] | undefined {
+  if (is.objectWithKeys(jsonData, ['urlPresets']) && is.array(jsonData.urlPresets)) {
+    const oldURLPresets = structuredClone(jsonData.urlPresets) as old_URLPreset;
+    const newURLPreset: URLPreset[] = oldURLPresets.map(({ enabled, alias, pathAndParams }) => {
+      const [target, search] = pathAndParams.split('?');
+      return { enabled, alias, target, search };
+    });
+    return newURLPreset;
+  }
+}
+
+type old_ProjectData = {
+  title: string;
+  description: string;
+  backstageUrl: string;
+  backstageInfo: string;
+  publicUrl: string;
+  publicInfo: string;
+  logo?: string; // is not present in old files
+  custom?: { title: string; value: string; url: string }[]; // is not present in old files
+};
+
+/**
+ * migrates a url presets from v3 to v4.0.0
+ * - `backstageUrl` -> `url`
+ * - `backstageInfo` -> `info`
+ * - drop `publicUrl`
+ * - drop `publicInfo`
+ * - ensure `logo`
+ * - ensure `custom`
+ */
+export function migrateProjectData(jsonData: object): ProjectData | undefined {
+  if (is.objectWithKeys(jsonData, ['project']) && is.object(jsonData.project)) {
+    const { title, description, backstageInfo, backstageUrl, logo, custom } = structuredClone(
+      jsonData.project,
+    ) as old_ProjectData;
+
+    const migrated: ProjectData = {
+      title,
+      description,
+      url: backstageUrl,
+      info: backstageInfo,
+      logo: logo ?? dbModel.project.logo,
+      custom: custom ?? dbModel.project.custom,
+    };
+    return migrated;
+  }
+}
+
+// old key -> new key
+type CustomFieldsTranslationTable = Map<string, string>;
+type old_CustomFields = Record<
+  string,
+  {
+    type: 'string' | 'image';
+    colour: string;
+    label: string;
+  }
+>;
+
+/**
+ * migrates a custom fields from v3 to v4.0.0
+ * - ensure correct case (TODO: could this be removed from the project parser)
```

Comment:
I believe so. the project parser should be able to assume that is parsing a v4 object